### ISSUE TITLE
Issues with version 3.4.1 to 3.4.3

### ISSUE THREAD

**Issue by insatmc** [SeleniumHQ/selenium#4297]
So I want to pass from version 3.4.0 to 3.4.3 but i get error 500 from the server.
Using 3.4.1 and 3.4.2 I get this error:
   
    `method_missing': undefined method `capabilities' for :chrome:Symbol (NoMethodError)
Usin 3.4.3 I get 
    
    `assert_ok': status code 500 (Selenium::WebDriver::Error::ServerError)

code: 

      @driver = Selenium::WebDriver.for(
          :remote,
          url: 'http://localhost:4444/wd/hub',
          desired_capabilities: :chrome)

Using: selenium-server-standalone-3.4.0.jar

**Comment by p0deje** [comment]
3.4.1 indeed had a bug, but 3.4.3 should be fine. 

> `assert_ok': status code 500 (Selenium::WebDriver::Error::ServerError)

This error means that something is wrong with standalone which is why it gives returns 500. Can you provide more output/log from standalone and full stacktrace of Ruby error?

**Comment by insatmc** [comment]
In the standalone the last thing in log is the node registration: 
   
    16:36:07.451 INFO - Registered a node http://10.92.5.121:5555 

The ruby log:

    /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/response.rb:72:in `assert_ok': status code 500 (Selenium::WebDriver::Error::ServerError)
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/response.rb:32:in `initialize'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/http/common.rb:83:in `new'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/http/common.rb:83:in `create_response'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/http/default.rb:107:in `request'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/http/common.rb:61:in `call'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/bridge.rb:170:in `execute'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/bridge.rb:103:in `create_session'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/bridge.rb:54:in `handshake'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/remote/driver.rb:33:in `new'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver/common/driver.rb:58:in `for'
	from /Users/mac/.rbenv/versions/2.3.4/lib/ruby/gems/2.3.0/gems/selenium-webdriver-3.4.3/lib/selenium/webdriver.rb:86:in `for'
	from test.rb:7:in `setup'
	from test.rb:18:in `run'
	from test.rb:23:in `<main>'

**Comment by p0deje** [comment]
Then I guess it happens on the node, can you get the output/log of the node?

**Comment by insatmc** [comment]
Nothing in the node also

    17:17:33.216 INFO - Registering the node to the hub: http://localhost:4444/grid/register
    17:17:33.263 INFO - The node is registered to the hub and ready to use
    17:17:38.401 INFO - SessionCleaner initialized with insideBrowserTimeout 0 and 
    clientGoneTimeout 1800000 polling every 180000

**Comment by p0deje** [comment]
I do believe something is wrong in your environment. It works perfectly fine when I try creating remote driver locally:

```
# hub output
21:24:38.063 INFO - Nodes should register to http://10.0.1.17:4444/grid/register/
21:24:38.063 INFO - Selenium Grid hub is up and running
21:25:05.508 INFO - Registered a node http://10.0.1.17:5555
21:25:37.164 INFO - Got a request to create a new session: Capabilities [{rotatable=false, nativeEvents=false, browserName=chrome, takesScreenshot=false, javascriptEnabled=true, version=, platform=ANY, cssSelectorsEnabled=true}]
21:25:37.165 INFO - Trying to create a new session on test slot {seleniumProtocol=WebDriver, browserName=chrome, maxInstances=5, platform=MAC}
```

```
# node output
21:25:05.508 INFO - The node is registered to the hub and ready to use
21:25:37.196 INFO - Executing: [new session: Capabilities [{rotatable=false, nativeEvents=false, browserName=chrome, takesScreenshot=false, javascriptEnabled=true, version=, platform=ANY, cssSelectorsEnabled=true}]])
21:25:37.222 INFO - Creating a new session for Capabilities [{rotatable=false, nativeEvents=false, browserName=chrome, takesScreenshot=false, javascriptEnabled=true, version=, platform=ANY, cssSelectorsEnabled=true}]
Starting ChromeDriver 2.30.477690 (c53f4ad87510ee97b5c3425a14c0e79780cdf262) on port 11127
Only local connections are allowed.
21:25:39.108 INFO - Detected dialect: OSS
21:25:39.134 INFO - Done: [new session: Capabilities [{rotatable=false, nativeEvents=false, browserName=chrome, takesScreenshot=false, javascriptEnabled=true, version=, platform=ANY, cssSelectorsEnabled=true}]]
21:25:39.193 INFO - Executing: [delete session: 28e8d47093fc16f879cd0c5a71f8e3c3])
21:25:39.269 INFO - Done: [delete session: 28e8d47093fc16f879cd0c5a71f8e3c3]
```

**Comment by p0deje** [comment]
Since there is no output on hub, can I be that you use incorrect URL for remote?

**Comment by insatmc** [comment]
Just uninstall the gem and install the version 3.4.0 it works so no i don't thing it's from the URL

**Comment by p0deje** [comment]
Interesting. The biggest change in 3.4.1 was implementation of dual-dialect protocol handshake. My guess is that it's related to #3808, so may I ask you to try to reproduce the issue:

1. Using selenium-server-standalone from latest master.
2. Using selenium-server-standalone from latest master with `-enablePassThrough` flag.

You'll need to compile the standalone jar by yourself. See [wiki page](https://github.com/SeleniumHQ/selenium/wiki/Building-WebDriver#building-selenium-server) for details on how to do that.

**Comment by barancev** [comment]
Should be fixed in version 3.5, please reopen the issue if it's not true.
